stl/inc/flat_map

```diff
@@ -0,0 +1,616 @@
+        return const_iterator { _Data.keys.cend(), _Data.values.end() };
+    }
+
+    _NODISCARD reverse_iterator rbegin() noexcept {
+        return _STD make_reverse_iterator(end());
+    }
+
+    _NODISCARD const_reverse_iterator rbegin() const noexcept {
+        return _STD make_reverse_iterator(end());
+    }
+
+    _NODISCARD reverse_iterator rend() noexcept {
+        return _STD make_reverse_iterator(begin());
+    }
+
+    _NODISCARD const_reverse_iterator rend() const noexcept {
+        return _STD make_reverse_iterator(begin());
+    }
+
+    _NODISCARD const_iterator cbegin() const noexcept {
+        return const_iterator {_Data.keys.cbegin(), _Data.values.cbegin()};
+    }
+
+    _NODISCARD const_iterator cend() const noexcept {
+        return const_iterator {_Data.keys.cend(), _Data.values.cend()};
+    }
+
+    _NODISCARD const_reverse_iterator crbegin() const noexcept {
+        return _STD make_reverse_iterator(cend());
+    }
+
+    _NODISCARD const_reverse_iterator crend() const noexcept {
+        return _STD make_reverse_iterator(cbegin());
+    }
+
+    // [container.reqmts] clear
+    void clear() noexcept {
+        _Data.keys.clear();
+        _Data.values.clear();
+    }
+
+    // [flat.map.capacity] Capacity
+    _NODISCARD_EMPTY_MEMBER bool empty() const noexcept {
+        return _Data.keys.empty();
+    }
+
+    _NODISCARD size_type size() const noexcept {
+        return _Data.keys.size();
+    }
+
+    _NODISCARD size_type max_size() const noexcept {
+        return _STD min(_Data.keys.max_size(), _Data.values.max_size());
+    }
+
+    // [flat.map.access] Access
+    template<class _K>
+    _NODISCARD mapped_type& operator[](_K&& _Key)
+        requires same_as<remove_cvref_t<_K>, key_type>
+            || (_Compare_is_transparent<key_compare> && convertible_to<_K&&, key_type>){
+        return try_emplace(_STD forward<_K>(_Key)).first->second;
+    }
+
+    _NODISCARD mapped_type& at(const key_type& _Key); // TODO: implement
+    _NODISCARD const mapped_type& at(const key_type& _Key) const; // TODO: implement
+    template <class _C>
+    _NODISCARD mapped_type& at(const _C& _Key)
+        requires _Compare_is_transparent<key_compare>; // TODO: implement
+    template <class _C>
+    _NODISCARD const mapped_type& at(const _C& _Key) const
+        requires _Compare_is_transparent<key_compare>; // TODO: implement
+
+    // [flat.map.modifiers] Modifiers
+    template <class... _Args_t>
+    _STD pair<iterator, bool> emplace(_Args_t&&... _Args) requires _STD is_constructible_v<value_type, _Args_t...> {
+        value_type _Val(_STD forward<_Args_t>(_Args)...);
+        auto _Key_It = _STD lower_bound(_Data.keys.begin(), _Data.keys.end(), _Val.first, _Key_compare);
+        iterator _It = begin() + _STD distance(_Data.keys.begin(), _Key_It);
+
+        if(_It != end() && _Key_equal(*(_It._Key_it), _Val.first)) {
+            // Already exists
+            return _STD make_pair(_It, false);

```

Comment:
The use of `make_pair` looks unnecessary to me.

Reply:
Removed all occurrences of `make_pair`.